src/Application/Actions/Donations/Create.php

```diff
@@ -279,4 +281,34 @@ private function removeSpecialChars(string $descriptor): string
     {
         return preg_replace('/[^A-Za-z0-9 ]/', '', $descriptor);
     }
+
+    /**
+     * It seems like just the *first* persist of a given donation needs to be retry-safe, since there is a small
+     * but non-zero minority of Create attempts at the start of a big campaign which get a closed Entity Manager
+     * and then don't know about the connected #campaign on persist and crash when RetrySafeEntityManager tries again.
+     */
+    private function persistDonationWithRetry(Donation $donation): void
+    {
+        \assert($this->entityManager instanceof RetrySafeEntityManager);
```

Comment:
any reason not to change the type on the actual property instead of asserting here? Fine if it's just for speed of implementation and we're going to change it later.

Reply:
No good reason, fixed in https://github.com/thebiggive/matchbot/pull/668/commits/0d135146007c36041ee2401d811034024798d7b5

Reply:
Looks like that would require setting an entry with `RetrySafeEntityManager::class` as the key in `dependancies.php` and configuring that in tests though. Or alternatively adding `\MatchBot\Application\Actions\Donations\Create` to `dependancies.php` and injecting deps manually to it.

Reply:
ah I see we already have an entry for RetrySafeEntityManager in dependencies.

Reply:
We already have it in `dependencies.php`, and I've just fixed the test